server/src/main/java/org/elasticsearch/plugins/stateless/ReloadingPlugin.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.plugins.stateless;
+
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.plugins.ReloadablePlugin;
+
+import java.util.List;
+
+/**
+ * A plugin that may receive a list of {@link ReloadablePlugin}s in order to
+ * call their {@link ReloadablePlugin#reload(Settings)} method.
+ */
+public interface ReloadingPlugin {
+
+    /**
+     * Provide a list of reloadable plugins.
+     * @param reloadablePlugins A list of plugins that this plugin may be able to reload
+     */
+    void setReloadablePlugins(List<ReloadablePlugin> reloadablePlugins);
```

Comment:
Consider insulating this, so that the plugin doesn't have direct access to all the other plugins, but instead gets a single ReloadablePlugin object which it can call, which internally wraps the list of plugins?

Reply:
I took a shot at this -- did I manage to do what you're describing?